docs/io/unified.rst

```diff
@@ -336,6 +354,73 @@ if both a width and precision are present they are both set in the TDISPn
 format.  A Python ``f`` or ``F`` map to TDISP F format.  The Python ``g`` or
 ``G`` map to TDISP G format.  The Python ``e`` and ``E`` map to TDISP E format.
 
+Masked columns
+^^^^^^^^^^^^^^
+
+Tables that contain `~astropy.table.MaskedColumn` columns can be written to
+FITS.  By default this will replace the masked data elements with certain
+sentinel values according to the FITS standard:
+
+- ``NaN`` for float columns
+- Value of ``TNULLn`` for integer columns, as defined by the column
+  ``fill_value`` attribute
+- Null string for string columns (not currently implemented)
+
+When the file is read back those elements are marked as masked in the returned
+table, but see `issue #4708 <https://github.com/astropy/astropy/issues/4708>`_
+for problems in all three cases.
+
+The FITS standard has a few limitations:
+
+- Not all data types are supported (e.g. logical / boolean)
+- Integer columns require picking one value as the NULL indicator.  If
+  all possible values are represented in valid data (e.g. an unsigned
+  int columns with all 256 possible values in valid data) then there
+  is no way to represent missing data.
+- The masked data values are permanently lost, precluding the possibility
+  of later unmasking the values.
+
+Astropy provides a work-around for this limitation that users can choose to
+use.  The key part using the ``serialize_method='data_mask'`` keyword argument
+when writing the table.  This tells the FITS writer to split each masked
+column into two separate columns, one for the data and one for the mask.
+When it gets read back that process is reversed and the two columns are
+merged back into one masked column.
+
+.. doctest-skip::
+
+  >>> from astropy.table.table_helpers import simple_table
+  >>> t = simple_table(masked=True)
+  >>> t['d'] = [False, False, True]
+  >>> t['d'].mask = [True, False, False]
+  >>> t
+  <Table masked=True length=3>
+    a      b     c     d
+  int64 float64 str1  bool
+  ----- ------- ---- -----
+     --     1.0    c    --
+      2     2.0   -- False
+      3      --    e  True
+
+.. doctest-skip::
```

Comment:
Perhaps we are skipping because this writes in the wrong place? I'm still a bit confused.

Reply:
IIRC (maybe not) some of the skipping was because PyYAML is not installed for docs generation / testing.  There was, at some point, a reason I put in the doctest-skip.  Also, in general I assume that docstring examples which demonstrate writing to a local file should be skipped.

Reply:
Let me ping @astrofrog and @bsipocz - my own sense would be to have as many doctests as feasible, and it would seem OK to require yaml for them (or treat it like scipy?)

Anyway, fine to treat this as out of scope for this PR!

Reply:
And basically I don't have a good relationship with sphinx and doc testing.  For instance in that `unified.rst` file I changed a line of code within a not-skipped section to obviously fail, and my local testing (e.g. `python setup.py test --package=io --args='-k rst'`) showed no problem.  So without having a way to reliably test locally (and not having time / desire to understand our sphinx testing), I don't want to risk burning more CI time.

Reply:
It's perfectly fine to require  all optional dependencies for the docs build. However doctests runs along with normal tests, so the incantation you need most probably is `.. doctest-requires:: pyyaml` before all code snippets that requires pyyaml (or any other optional dependencies).

Reply:
Thanks @bsipocz, nice to know!  Since this is passing tests now, maybe I'll let this slide for now and clean it up later in some other context to avoid burning CI.

Reply:
Sure. That docs cleanup seems to me an ideal first PR issue, so if you could open a reminder for it we can label it accordingly to be picked up by someone else.